terraform/gcp/modules/cloud_run_v2/run_v2_service.tf

```diff
@@ -0,0 +1,88 @@
+data "google_project" "project" {
+}
+
+locals {
+  project = data.google_project.project.project_id
+}
+
+
+resource "google_cloud_run_v2_service" "with_vpc_access" {
+  count = length(var.vpc_connector) > 0 ? 1 : 0
+  name     = var.service_name
+  location = var.location
+  ingress = "INGRESS_TRAFFIC_ALL"
+
+  template {
+    scaling {
+      min_instance_count = var.min_instance_count
+      max_instance_count = var.max_instance_count
+
+
+    }
+
+    containers {
+      image = var.img_url
+      ports {
+        container_port = var.container_port
+      }
+      
+
+
+      dynamic "env" {
+        for_each = var.container_env
+        content {
+          name = env.key
+          value = env.value
+        }
+        
+      }
+    }
+    vpc_access{
```

Comment:
cant the vpc_access be also kept in a dynamic block? if yes, you dont need 2 separate resources.